agent/src/beerocks/slave/ap_manager_thread.cpp

```diff
@@ -401,6 +401,35 @@ void ap_manager_thread::after_select(bool timeout)
                 now + std::chrono::seconds(HEARTBEAT_NOTIFICATION_DELAY_SEC);
         }
     }
+
+    // check if any client disallow period has expired
+    // in case it did we need to removed the client from blacklist
+    if (!m_disallowed_client_timeouts.empty()) {
+        // map key holds the disallow timout
+        // since its an ordered map check always the first element (lowest value)
+        auto it                          = m_disallowed_client_timeouts.begin();
```

Comment:
If there are multiple timeouts running, it is possible and even likely (*) that more than one expires at the same time. So you really have to iterate here.

Because of that, I'd think it's easier to use a std::vector and just iterate over everything.

(*) Likely because: either we're steering just a single client, and then there will be only one element in this list. Or we're steering all clients away, e.g. because we're going to switch channels, and in that case all the timeouts will expire at the same time. So the only situation where we have multiple elements in the list that don't expire at the same time is when a second client is steered a few seconds after the first one. It happens, but not very likely.